spring-integration-zeromq/src/main/java/org/springframework/integration/zeromq/outbound/ZeroMqMessageHandler.java

```diff
@@ -214,6 +295,8 @@ public String getComponentType() {
 	@Override
 	protected void onInit() {
 		super.onInit();
+		Assert.state(this.connectUrl == null || this.bindPort.get() == 0,
+				"Only one of the 'connectUrl' or 'bindPort' must be provided");
```

Comment:
If we remove those new setters, this condition won't be necessary.
So, you would only have a choice to give `connectUrl` or `port` via those constructors.